yt_dlp/extractor/mocha.py

```diff
@@ -0,0 +1,65 @@
+from .common import InfoExtractor
+from ..utils import int_or_none, traverse_obj
+
+
+class MochaVideoIE(InfoExtractor):
+    _VALID_URL = r'https?://video.mocha.com.vn/(?P<video_slug>[\w-]+)'
+    _TESTS = [{
+        'url': 'http://video.mocha.com.vn/chuyen-meo-gia-su-tu-thong-diep-cuoc-song-v18694039',
+        'info_dict': {
+            'id': '18694039',
+            'title': 'Chuyện mèo giả sư tử | Thông điệp cuộc sống',
+            'ext': 'mp4',
+            'view_count': int,
+            'like_count': int,
+            'dislike_count': int,
+            'display_id': 'chuyen-meo-gia-su-tu-thong-diep-cuoc-song',
+            'thumbnail': 'http://mcvideomd1fr.keeng.net/playnow/images/20220505/ad0a055d-2f69-42ca-b888-4790041fe6bc_640x480.jpg',
+            'description': '',
+            'duration': 70,
+            'timestamp': 1652254203,
+            'upload_date': '20220511',
+            'comment_count': int,
+            'categories': ['Kids']
+        }
+    }]
+
+    def _real_extract(self, url):
+        video_slug = self._match_valid_url(url).group('video_slug')
+        json_data = self._download_json(
+            'http://apivideo.mocha.com.vn:8081/onMediaBackendBiz/mochavideo/getVideoDetail',
+            video_slug, query={'url': f'{url}', 'token': ''})
+
+        video_url = traverse_obj(json_data, ('data', 'videoDetail', ('list_resolution', 'original_path')))
```

Comment:
```suggestion
        video_urls = traverse_obj(json_data, ('data', 'videoDetail', ('list_resolution', 'original_path'))) or []
```

Reply:
any good name will work.
original_video_paths, original_paths, video_paths, etc.

video_url is a bad choice bcoz it implies it is a string

Reply:
Sorry, but i didn't understand what you meant. At [this comment](https://github.com/yt-dlp/yt-dlp/pull/4213#discussion_r910489239), i was asking help to fix `traverse_obj` didn't 'catch' `original_path` in return. It just return `list_original`. I want to include `original_path` in `video_urls` as well

Reply:
oh okay, I thought u meant the variable name is not good. 

I think it can be fixed by doing 
` video_urls = traverse_obj(json_data, ('list_resolution', 'original_path')) or []`

```
''' Traverse nested list/dict/tuple
    @param path_list        A list of paths which are checked one by one.
                            Each path is a list of keys where each key is a:
                              - None:     Do nothing
                              - string:   A dictionary key
                              - int:      An index into a list
                              - tuple:    A list of keys all of which will be traversed
```

But then i haven't used this before so I might be wrong here, @pukkandan is better person to answer

Reply:
AFAICT,
He has a dict like
```
main_dict = {
'data': {
'videoDetail': {
'list_resolution': stuff,
'original_path': stuff

}
}
}
```

Now he wants traverse_obj to return `stuff` both `list_resolution` and `original_path` from `json_data`

`json_data = main_dict['data']['videoDetail']`

Reply:
```
>>> from yt_dlp.utils import traverse_obj
>>> json_data = {'list_resolution': ['1', '2'], 'original_path': '3'}
>>> traverse_obj(json_data, (('list_resolution', 'original_path'), ))
[['1', '2'], '3']
```
But you don't really even need `traverse_obj` for that
```
>>> json_data.get('list_resolution') or [], json_data.get('original_path')
(['1', '2'], '3')
```
Or maybe, if we wants it as a single list:
```
>>> (json_data.get('list_resolution') or []) + [json_data.get('original_path')]
['1', '2', '3']
```

Reply:
Thanks both of you, it fixed in https://github.com/yt-dlp/yt-dlp/pull/4213/commits/4decae0936248dc75c087bee0a99c7de6874095a